test/asn1_time_test.c

```diff
@@ -443,6 +443,26 @@ static int convert_asn1_to_time_t(int idx)
     return 1;
 }
 
+/*
+ * this test is here to exercise ossl_asn1_time_from_tm
+ * with an integer year close to INT_MAX.
+ */
+static int convert_tm_to_asn1_time(void)
+{
+  /* we need 64 bit time_t */
+#if ( (ULONG_MAX >> 31) >> 31 ) >= 1
+    time_t t;
+    ASN1_TIME* at;
+
+    TEST_int_ge(sizeof(time_t) * CHAR_BIT, 64);
```

Comment:
Probably limits.h needs to be included for `CHAR_BIT`?

This should not be enclosed in `TEST_int_ge` I think. This will cause failure errors in the output if we don't have 64 bit time. Probably there should just be an "if" here and we skip the remaining code if it fails.

Reply:
fixed both limits.h and switch to if.